tests/test_integrals.py

```diff
@@ -938,93 +936,66 @@ class TestBounceQuadrature:
-        v = 7
-        z1 = -np.pi / 2 * v
-        z2 = -z1
-        knots = np.linspace(z1, z2, 50)
-        pitch_inv = 1 - 50 * jnp.finfo(jnp.array(1.0).dtype).eps
-        b = np.clip(np.sin(knots / v) ** 2, 1e-7, 1)
-        db = np.sin(2 * knots / v) / v
-        data = {"B^zeta": b, "B^zeta_z|r,a": db, "|B|": b, "|B|_z|r,a": db}
-
-        if is_strong:
-            integrand = lambda B, pitch: 1 / jnp.sqrt(1 - m * pitch * B)
-            truth = v * 2 * ellipkm1(p)
-        else:
-            integrand = lambda B, pitch: jnp.sqrt(1 - m * pitch * B)
-            truth = v * 2 * ellipe(m)
+        v = 3
+        knots = np.linspace(-np.pi / 2 * v, np.pi / 2 * v, 50)
+        B = np.sin(knots / v) ** 2 + 1
+        dB_dz = np.sin(2 * knots / v) / v
         bounce = Bounce1D(
             Grid.create_meshgrid([1, 0, knots], coordinates="raz"),
-            data,
-            quad,
-            automorphism,
+            data={"B^zeta": B, "B^zeta_z|r,a": dB_dz, "|B|": B, "|B|_z|r,a": dB_dz},
+            quad=quad,
+            automorphism=automorphism,
             check=True,
         )
-        points = bounce.points(pitch_inv, num_well=1)
-        np.testing.assert_allclose(points[0], z1)
-        np.testing.assert_allclose(points[1], z2)
-        result = bounce.integrate(
-            integrand, pitch_inv, points=points, check=True, plot=True
+
+        m = 1 - 1e-3
+        # integral(integrand) = truth iff pitch_inv = 2
+        pitch_inv = 2 - 1e-12
+        k = pitch_inv * m  # m = k * pitch
+        if is_strong:
+            integrand = lambda data, B, pitch: 1 / jnp.sqrt(1 - k * pitch * (B - 1))
+            truth = v * 2 * ellipkm1(1 - m)
+        else:
+            integrand = lambda data, B, pitch: jnp.sqrt(1 - k * pitch * (B - 1))
+            truth = v * 2 * ellipe(m)
+        np.testing.assert_allclose(
+            bounce.integrate(integrand, pitch_inv, check=True, plot=False).sum(),
+            truth,
+            rtol=1e-4,
         )
-        assert np.count_nonzero(result) == 1
-        np.testing.assert_allclose(result.sum(), truth, rtol=1e-4)
 
     @staticmethod
     @partial(np.vectorize, excluded={0})
     def _adaptive_elliptic(integrand, k):
         a = 0
         b = 2 * np.arcsin(k)
-        return integrate.quad(integrand, a, b, args=(k,), points=b)[0]
+        return integrate.quad(
+            integrand, a, b, args=(k,), points=b, epsrel=1e-10, epsabs=1e-10
+        )[0]
 
     @staticmethod
     def _fixed_elliptic(integrand, k, deg):
         k = np.atleast_1d(k)
-        a = np.zeros_like(k)
+        a = -2 * np.arcsin(k)
         b = 2 * np.arcsin(k)
         x, w = get_quadrature(leggauss(deg), (automorphism_sin, grad_automorphism_sin))
         Z = bijection_from_disc(x, a[..., np.newaxis], b[..., np.newaxis])
         k = k[..., np.newaxis]
-        quad = integrand(Z, k).dot(w) * grad_bijection_from_disc(a, b)
+        quad = integrand(Z, k).dot(w) * grad_bijection_from_disc(a, b) / 2
```

Comment:
because I changed the integration from [0, 2 arcsin] to [-2 arcsin, 2 arcsin]. This is because the modified chebyshev quadrature I use only works with singularity at both ends, while the modified gass legendre handles anything (no singularity at 0)